packages/manager/modules/bm-server-components/src/server-reboot/server-reboot.controller.js

```diff
@@ -0,0 +1,65 @@
+export default class DedicatedServerRebootCtrl {
+  /* @ngInject */
+  constructor($q, $scope, $stateParams, $state, $translate, Alerter, Server) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$stateParams = $stateParams;
+    this.$state = $state;
+    this.$translate = $translate;
+    this.Alerter = Alerter;
+    this.Server = Server;
+  }
+
+  $onInit() {
+    this.server = null;
+    this.isLoading = false;
+    this.isRebooting = false;
+
+    this.isLoading = true;
+    return this.Server.getSelected(this.$stateParams.productId)
```

Comment:
Is it possible to move the getSelected() to the service inside bm-server-component ?

Reply:
I created the services in bm-server-component sever-reboot. I also remove the first call that recall the detail server because the server object is already exist so no need.